### ISSUE TITLE
Cancelling auto-fill does not work

### ISSUE THREAD

**Issue by markywarky** [bitwarden/clients#5899]
### Steps To Reproduce

1. Navigate browser to a page with a form whose domain name is different to the URI of the saved login, e.g. https://mybmt.bmtqs.com.au/login.
2. Press Cmd-Shift-L to auto-fill.
3. A warning is displayed: "The form is hosted by a different domain than the URI of your saved login. Choose OK to auto-fill anyway, or Cancel to stop."
<img width="604" alt="Screenshot 2023-07-26 at 5 44 38 pm" src="https://github.com/bitwarden/clients/assets/76718662/49a70ec4-a8fa-4795-bc2c-f08692e4ef00">
4. Click on Cancel.

### Expected Result

Warning should disappear and no form filling should take place.

### Actual Result

Form is auto-filled with vault credentials for that URI.

### Screenshots or Videos

<img width="604" alt="Screenshot 2023-07-26 at 5 44 38 pm" src="https://github.com/bitwarden/clients/assets/76718662/56f0ced4-9a00-402d-b9f2-028a3343d8c4">

### Operating System

macOS

### Operating System Version

Ventura 13.4 (22F66)

### Web Browser

Safari, Firefox

### Browser Version

Safari 16.5, Firefox 115

### Build Version

Safari 18615.2.9.11.4, Firefox 115.0.2 (64-bit)

### Issue Tracking Info

- [X] I understand that work is tracked outside of Github. A PR will be linked to this issue should one be opened to address it, but Bitwarden doesn't use fields like "assigned", "milestone", or "project" to track progress.

**Comment by Krychaz** [comment]
Hi there,

Thank you for your report! 

I was able to reproduce this issue, and I have flagged this to our engineering team.

If you wish to add any further information/screenshots/recordings etc., please feel free to do so at any time - our engineering team will be happy to review these.

Thanks once again!

**Comment by markywarky** [comment]
This problem is still occurring on plugin version 2023.12.0, server version 2024.1.0.

**Comment by cagonzalezcs** [comment]
Hey @markywarky 

Thanks for providing this bug report. So in reviewing the issue, I found that cancelling the iframe prompt indeed impedes the filling of credentials. 

The site that you mentioned has an iframe which is visibly hidden. That iframe has a login form which triggers our warning when autofilling into the web page. Please see the video below to see what is happening:

https://github.com/bitwarden/clients/assets/16629865/97c7be0c-ab69-47c6-9fdf-1cb2391ed759

Now, the bigger question I'm asking myself at the moment is "Why are we attempting to fill these credentials when they are present within a visibly hidden iframe element." The form element within the iframe itself is even set with a `display: none` inline CSS value. 

I'll investigate this further when time allows.

**Comment by markywarky** [comment]
Nice work, @cagonzalezcs. 

You're discussing why the warning is triggered in this situation. Great question. But regardless of the answer, don't you think that auto-filling should stop if the user clicks Cancel?

**Comment by cagonzalezcs** [comment]
@markywarky 

The way our script logic is implemented, auto-fill does stop if the user clicks Cancel, but you have to keep in mind the question "where are we attempting to trigger this autofill?"

In the example above, autofill gets triggered in at least two different places. It's triggered at the top level of the page (whose `window.location` is `https://mybmt.bmtqs.com.au/login`), as well as the iframe that the page contains (which has a `src` of `https://directinfo.com.au/home/logoutuser`).

Based on [our insert logic](https://github.com/bitwarden/clients/blob/03c6037a9c9b40f8584f34071bbf8eb7dec20ed1/apps/browser/src/autofill/services/insert-autofill-content.service.ts#L40), the top frame doesn't break any of the rules for filling. It won't send the message that you see warning users that you are filling in a different host context, and won't allow you to "cancel" the auto-fill. 

However, when we target a tab to fill, we are triggering a fill in any frame contexts that contain fillable fields. We do this due to how some websites setup their form field element, for instance `icloud.com` which hosts their login form at `idmsa.apple.com`. (Note: There's an internal reason why this doesn't trigger the same kind of warning as it does on your example website. We have an internal list of equivalent domains that we use to validate if this kind of warning is needed). 

So with that said, because the iframe within the `mybmt.bmtqs.com` contains fields, we end up triggering autofill within that iframe with your selected cipher. It triggers on both the form at `mybmt.bmtqs.com` and within the iframe at `directinfo.com.au`.

The warning comes into play here, however. Since `directinfo.com.au` is a different domain from `mybmt.bmtqs.com`, and the cipher only has information for `mybmt.bmtqs.com`, we trigger a warning to the user asking if they are really sure they would like to fill their credentials within the iframe pointed at `directinfo.com.au`. When the user clicks "cancel" it stops that fill from happening in `directinfo.com.au`, but it won't stop that fill from happening within form that is at the top level `mybmt.bmtqs.com`.

**Comment by markywarky** [comment]
Yeah, it's an interesting issue, @cagonzalezcs. No doubt you're correct wth your technical details, and it's good to know that BitWarden is doing the right thing. But most users don't know about the HTML, they only see filling happening after they have cancelled it.

**Comment by closebot-bw** [comment]
⚠️ **Stale Issue Notice**

This issue has been automatically marked as stale due to inactivity. It will be closed in **2 weeks** (October 15, 2025) if no further activity occurs.

If this issue is still relevant and you would like to keep it open, please:
- Comment on this issue to show continued interest
- Provide any additional information or updates
- Confirm that the issue still exists in the latest version

Thank you for your contribution to this project! 🙏

**Comment by closebot-bw** [comment]
🔒 **Issue Closed Due to Inactivity**

This issue has been automatically closed due to lack of activity for an extended period. We periodically review and close inactive issues to help maintain our issue tracker and focus on current priorities.

**If this issue is still relevant:**
- Please create a new issue with updated information
- Include steps to reproduce the problem if it's a bug report
- Mention if this issue still occurs in the latest version

Thank you for your contribution to this project. Your feedback helps us improve! 🙏
